### PR TITLE
Z algorithm in C

### PR DESCRIPTION
**Pull request by SangeetaMishra**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#417

### CODE REVIEW

**Review by iamrajiv — APPROVED** [review]

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

File: C/cp/zalgorithm.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        }
+    }
+    return 0;
+}
+
+/*
+Time Complexity: O(length of text + length of pattern)
+
+Input
+AABACAAFGAABAABA
```
Comment (atarax665) [inline C/cp/zalgorithm.c]: Sample input/output should be same as what we will get when the program is run in IDE.
eg- ```Enter the text: AABACAAFGAABAABA```

**Review by atarax665 — CHANGES_REQUESTED** [review]
Format your code using https://www.techiedelight.com/tools/clike

File: C/cp/zalgorithm.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	char text[100], pattern[100];
+	printf("Enter the text: ");
+	scanf("%s", text);
+	printf("\nEnter the pattern: ");
+	scanf("%s", pattern);
+	int m = strlen(pattern);
+	strcat(pattern, "$");
+	strcat(pattern, text);
+	int n = strlen(pattern);
+	int z[n] = {};
```
Comment (atarax665) [inline C/cp/zalgorithm.c]:
```suggestion
	int z[n];
```

**Review by atarax665 — APPROVED** [review]
Looks good to me now. DO this small change and it will be good to merge.

**Pull request merged**